add extra method for use by unregister RPC api

arnej27959 opened the pull request:
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@havardpe please review

arnej27959 committed:
add extra method for use by unregister RPC api

havardpe commented on slobrok/src/vespa/slobrok/server/local_rpc_monitor_map.h:
```diff
@@ -126,6 +129,9 @@ class LocalRpcMonitorMap : public MapListener,
                   std::unique_ptr<ScriptCommand> inflight);
 
+    /** for use by unregister API, returns true on success */
+    bool removeLocal(const ServiceMapping &mapping);
```
consider using wouldConflict in the consensus map instead to get this bool (is this a 'clean' remove)

havardpe reviewed (commented):
might need to clear ignoreFirstOk when reusing the per service object.

arnej27959 commented:
@havardpe please take another look

havardpe commented on slobrok/src/vespa/slobrok/server/local_rpc_monitor_map.cpp:
```diff
@@ -93,6 +93,7 @@ void LocalRpcMonitorMap::addLocal(const ServiceMapping &mapping,
         if (exists.spec == mapping.spec) {
             LOG(debug, "added mapping %s->%s was already present",
                 mapping.name.c_str(), mapping.spec.c_str());
+            // clear exists.ignoreFirstOk ?
```
another question; should we really return 'ok' when we known the service is in the process of being removed?

arnej27959 committed:
handle addLocal for down services better

* if we already have a registration in progress, hook the second
  one to the first so they get the same answer at the same time
* if we think the service is down, restart its monitoring to
  avoid getting stale information, and add the inflight hook so we get an
  answer when the new ping returns.

arnej27959 committed:
remove ignoreFirstOk flag

* instead of ignoring one ping, restart monitoring of the mapping
  (without hurry flag).
